apps/src/codebridge/hooks/useInitialSources.ts

```diff
@@ -5,22 +5,33 @@ import {getAppOptionsEditBlocks} from '@cdo/apps/lab2/projects/utils';
 import {ProjectSources} from '@cdo/apps/lab2/types';
 import {useAppSelector} from '@cdo/apps/util/reduxHooks';
 
+// Hook for getting the initial sources for the current level.
+// If we are in start mode, we show the start sources or the default sources
+// if the level has no start sources.
+// If the user has a project, we prioritize that over the start sources. If there is
+// no project or start sources, we use the default sources.
 export const useInitialSources = (defaultSources: ProjectSources) => {
-  const labState = useAppSelector(state => state.lab);
+  const labInitialSources = useAppSelector(state => state.lab.initialSources);
+  const levelStartSource = useAppSelector(
+    state => state.lab.levelProperties?.source
```

Comment:
Just confirming that `levelProperties.source` is the right place for starter code to go? I know that in legacy labs we use either `start_blocks` or `start_sources` (maybe only for Javalab?). Music lab is a weird special case because the starter code is in `levelData` but I think it's called `startSources` there too? (either way, I'd like to move it out of `levelData` at some point anyway)